components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOFacade.java

```diff
@@ -0,0 +1,418 @@
+import org.wso2.carbon.database.utils.jdbc.exceptions.TransactionException;
+import org.wso2.carbon.identity.action.management.constant.error.ErrorMessage;
+import org.wso2.carbon.identity.action.management.dao.ActionManagementDAO;
+import org.wso2.carbon.identity.action.management.exception.ActionDTOModelResolverClientException;
+import org.wso2.carbon.identity.action.management.exception.ActionDTOModelResolverException;
+import org.wso2.carbon.identity.action.management.exception.ActionMgtClientException;
+import org.wso2.carbon.identity.action.management.exception.ActionMgtException;
+import org.wso2.carbon.identity.action.management.exception.ActionMgtServerException;
+import org.wso2.carbon.identity.action.management.model.Action;
+import org.wso2.carbon.identity.action.management.model.ActionDTO;
+import org.wso2.carbon.identity.action.management.model.AuthProperty;
+import org.wso2.carbon.identity.action.management.model.Authentication;
+import org.wso2.carbon.identity.action.management.model.EndpointConfig;
+import org.wso2.carbon.identity.action.management.service.ActionDTOModelResolver;
+import org.wso2.carbon.identity.action.management.util.ActionDTOBuilder;
+import org.wso2.carbon.identity.action.management.util.ActionManagementExceptionHandler;
+import org.wso2.carbon.identity.action.management.util.ActionSecretProcessor;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.secret.mgt.core.exception.SecretManagementException;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Facade class for Action Management DAO.
+ * ActionManagementDAOFacade is responsible for handling external service integrations.
+ */
+public class ActionManagementDAOFacade implements ActionManagementDAO {
+
+    private static final Log LOG = LogFactory.getLog(ActionManagementDAOFacade.class);
+
+    private final ActionManagementDAO actionManagementDAO;
+    private final ActionSecretProcessor actionSecretProcessor;
+
+    public ActionManagementDAOFacade(ActionManagementDAO actionManagementDAO) {
+
+        this.actionManagementDAO = actionManagementDAO;
+        this.actionSecretProcessor = new ActionSecretProcessor();
+    }
+
+    @Override
+    public void addAction(ActionDTO actionDTO, Integer tenantId) throws ActionMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            jdbcTemplate.withTransaction(template -> {
+                ActionDTOBuilder actionDTOBuilder = new ActionDTOBuilder(actionDTO);
+                // Encrypt authentication secrets
+                encryptAddingAuthSecrets(actionDTOBuilder);
+                // Resolve action properties
+                ActionDTO resolvedActionDTO = getActionDTOWithResolvedAddingProperties(actionDTOBuilder.build(),
```

Comment:
Shall we rename this method for a better name ?
May be -> getResolvedActionDTOWithPropertiesForAddOperation or getResolvedActionDTOForAddOperation